src/pages/steps/AdditionalInformationStep/PriceInformation.tsx

```diff
@@ -81,26 +83,49 @@ const shouldHaveAName = (value: any): boolean => {
   return !!value[i18n.language];
 };
 
+yup.addMethod(yup.array, 'uniqueName', function () {
+  return this.test('uniqueName', function (prices) {
+    const priceNames = prices.map((item) => item.name[i18n.language]);
+    const errors = priceNames.map((priceName, index) => {
+      const indexOf = priceNames.indexOf(priceName);
+      if (indexOf !== -1 && indexOf !== index) {
+        return this.createError({
+          path: `${this.path}.${index}`,
+          message: i18n.t(
+            'create.additionalInformation.price_info.duplicate_name_error',
+            { priceName },
+          ),
+        });
+      }
+    });
+
+    return new ValidationError(errors.filter(Boolean));
+  });
+});
+
 const schema = yup
   .object()
   .shape({
-    rates: yup.array().of(
-      yup.object({
-        name: yup
-          .object({
-            nl: yup.string(),
-            fr: yup.string(),
-            en: yup.string(),
-            de: yup.string(),
-          })
-          .test(`name-is-required`, 'name is required', shouldHaveAName)
-          .test(`name-is-not-uitpas`, 'should not be uitpas', isNotUitpas)
-          .required(),
-        category: yup.string(),
-        price: yup.string().matches(PRICE_REGEX).required(),
-        priceCurrency: yup.string(),
-      }),
-    ),
+    rates: yup
+      .array()
+      .of(
+        yup.object({
+          name: yup
+            .object({
+              nl: yup.string(),
+              fr: yup.string(),
+              en: yup.string(),
+              de: yup.string(),
+            })
+            .test(`name-is-required`, 'name is required', shouldHaveAName)
+            .test(`name-is-not-uitpas`, 'should not be uitpas', isNotUitpas)
+            .required(),
+          category: yup.string(),
+          price: yup.string().matches(PRICE_REGEX).required(),
+          priceCurrency: yup.string(),
+        }),
+      )
+      .uniqueName(),
```

Comment:
```suggestion
      .test("uniqueName", yourValidationFunction);
```

I prefer just chaining a test function to it instead of adding a method to `yup.array`

Reply:
Ah nice that's better indeed, plus better addition/deletion ratio :p